packages/openapi-v3/src/controller-spec.ts

```diff
@@ -186,6 +186,15 @@ function resolveControllerSpec(constructor: Function): ControllerSpec {
     }
 
     operationSpec['x-operation-name'] = op;
+    operationSpec['x-controller-name'] =
+      operationSpec['x-controller-name'] || constructor.name;
+
+    if (operationSpec.operationId == null) {
+      operationSpec.operationId =
+        operationSpec['x-controller-name'] +
+        '$' +
```

Comment:
Ditto, have you considered using `.` instead?

Reply:
See above.